src/conn.c

```diff
@@ -4030,6 +4031,43 @@ natsConnection_GetLastError(natsConnection *nc, const char **lastError)
     return s;
 }
 
+natsStatus
+natsConnection_ReadLastError(natsConnection *nc, char *buf, size_t n)
+{
+    natsStatus  s;
+
+    if (nc == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    natsConn_Lock(nc);
+
+    s = nc->err;
+    if (s == NATS_OK)
+        nc->errStr[0] = '\0';
+    else if (nc->errStr[0] == '\0')
+        snprintf(nc->errStr, sizeof(nc->errStr), "%s", natsStatus_GetText(s));
+
+    if ((buf != NULL) && (n > 0))
+    {
+        size_t errLen = strlen(nc->errStr);
+        strncpy(buf, nc->errStr, n-1);
```

Comment:
`strncpy` I believe will add with 0 if the source length is lower than the `n-1` you provide. I would have probably check the min of `errLen` and `n` to know how much to copy, but take into account that we would need to add the terminating `\0`.

Reply:
Oh (regardless!) this was a mistake; better style would be to pass the (ptr, n) around "as is". I'll replace it with a `memcpy` since the lengths are known.